### PR TITLE
Skill rewrite & cascade

### PR DESCRIPTION
**Pull request by FKD13**

### CODE REVIEW

**Comment by FKD13** [comment]
Ready for review! Note this brach is branched from #344. To keep things tidy in that pr we might want to merge that first!

**Review by stijndcl — CHANGES_REQUESTED** [review]
One more note: there's a couple of fields that are now nullable (as requested by Miet), but I've noticed these aren't actually nullable in the Pydantic API schema's (eg. the coach/drafter/user in a Suggestion). That might cause a problem. Could you double-check on those?

File: backend/src/app/routers/editions/projects/students/projects_students.py [hunk 1]
```diff
@@ -5,52 +5,66 @@
...
 
-@project_students_router.patch("/{student_id}", status_code=status.HTTP_204_NO_CONTENT, response_class=Response,
-                               dependencies=[Depends(get_latest_edition)])
-async def change_project_role(student_id: int, input_sr: InputStudentRole, db: Session = Depends(get_session),
-                              project: Project = Depends(get_project), user: User = Depends(require_coach)):
+@project_students_router.patch(
+    "/{student_id}",
+    status_code=status.HTTP_204_NO_CONTENT,
+    response_class=Response,
+    dependencies=[Depends(get_latest_edition), Depends(get_project_role)]
```
[the remark this answers is not in the document]
Comment (FKD13) [reply backend/src/app/routers/editions/projects/students/projects_students.py:39]: removed!

File: backend/src/app/routers/editions/projects/projects.py [hunk 2]
```diff
@@ -5,36 +5,43 @@
...
-    """
-    return logic.create_project(db, edition,
-                                input_project)
+@projects_router.post(
+    "/",
+    status_code=status.HTTP_201_CREATED,
+    response_model=Project,
+    dependencies=[Depends(require_admin)]
+)
+async def create_project(
```
[the remark this answers is not in the document]
Comment (FKD13) [reply backend/src/app/routers/editions/projects/projects.py:39]: We did discuss this in chat, we agreed on creating the project roles with separate requests

**Comment by FKD13** [comment]
Ready for review again!

**Comment by stijndcl** [comment]
(You forgot the change in the message of my PR)

**Review by stijndcl — APPROVED** [review]

**Pull request merged**